benchmark/index.js

```diff
@@ -0,0 +1,180 @@
+const getTestData = (intgList, fileNameSuffix) => {
+  const intgTestSet = {};
+  intgList.forEach(intg => {
+    try {
+      intgTestSet[intg] = JSON.parse(
+        fs.readFileSync(
+          path.join(testDataDir, `${intg}${fileNameSuffix}.json`),
+          {
+            encoding: "utf-8"
+          }
+        )
+      );
+    } catch (err) {
+      logger.error(`Unable to load the data for: ${intg}`);
+      logger.error(`Raw error: ${err}`);
+    }
+  });
+  return intgTestSet;
+};
+
+const destinationsList = cmdOpts.destinations
+  .split(",")
+  .map(x => x.trim())
+  .filter(x => x !== "");
+logger.info("Destinations selected: ", destinationsList);
+const destTestSet = getTestData(destinationsList, "_input");
+const nativeDestHandlers = {};
+destinationsList.forEach(dest => {
+  nativeDestHandlers[dest] = versionedRouter.getDestHandler("v0", dest);
+});
+
+const sourcesList = cmdOpts.sources
+  .split(",")
+  .map(x => x.trim())
+  .filter(x => x !== "");
+logger.info("Sources selected: ", sourcesList);
+const srcTestSet = getTestData(sourcesList, "_source");
+
+async function run() {
+  for (const dest in destTestSet) {
+    for (const td in destTestSet[dest]) {
+      const curTcData = destTestSet[dest][td];
+      let tcInput;
+      let tcDesc;
+      if (
+        "description" in curTcData &&
+        "input" in curTcData &&
+        "output" in curTcData
+      ) {
+        tcInput = curTcData.input;
+        tcDesc = `Destination - ${dest} - "${curTcData.description}"`;
+      } else {
+        tcInput = curTcData;
+        tcDesc = `Destination - ${dest} - "${td}"`;
+      }
+
+      await runDataset(tcDesc, tcInput, dest);
+    }
+  }
+
+  for (const src in srcTestSet) {
+    for (const td in srcTestSet[src]) {
+      const curTcData = srcTestSet[src][td];
+      let tcInput;
+      let tcDesc;
+      if (
+        "description" in curTcData &&
+        "input" in curTcData &&
+        "output" in curTcData
+      ) {
+        tcInput = curTcData.input;
+        tcDesc = `Source - ${src} - "${curTcData.description}"`;
+      } else {
+        tcInput = curTcData;
+        tcDesc = `Source - ${src} - "${td}"`;
+      }
+
+      await runDataset(tcDesc, tcInput, src);
+    }
+  }
+}
+
+async function runDataset(desc, input, intg) {
+  const suiteName = desc;
+  logger.info("==========================================");
+  logger.info(suiteName);
+  logger.info("==========================================");
+  const results = {};
+  const suite = new Benchmark(suiteName, "Operations");
+
+  suite.add("native", async function() {
+    try {
+      await versionedRouter.handleV0Destination(
+        nativeDestHandlers[intg].process,
+        intg,
+        input,
+        TRANSFORMER_METRIC.ERROR_AT.PROC
+      );
+    } catch (err) {
+      // logger.info(err);
+      // Do nothing
+    }
+  });
+
+  suite.add("CDK 2.0", async function() {
+    try {
+      await cdkV2Handler.processCdkV2Workflow(
```

Comment:
As we discussed please create workflow engine instance before executing the benchmarking

Reply:
Addressed.